frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java

```diff
@@ -120,41 +129,76 @@ public void run(ProgressIndicator indicator) {
   @NotUIThreadSafe
   private void deleteBranchIfRequired(AnActionEvent anActionEvent, String branchName) {
     var gitRepository = getSelectedGitRepository(anActionEvent);
-
-    if (gitRepository.isDefined()) {
+    var project = getProject(anActionEvent);
+    var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
+        .map(b -> b.equals(branchName))
+        .getOrElse(true);
+
+    if (slidOutBranchIsCurrent) {
+      LOG.debug("Skipping (optional) local branch deletion because it is equal to current branch");
+      getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+      VcsNotifier.getInstance(project)
+          .notifySuccess(
+              format(
+                  getString("action.GitMachete.BaseSlideOutBranchAction.notification.title.slide-out-success.of-current"),
+                  branchName));
+
+    } else if (gitRepository.isDefined()) {
       var root = gitRepository.get().getRoot();
-      var project = getProject(anActionEvent);
       var shallDeleteLocalBranch = getDeleteLocalBranchOnSlideOutGitConfigValue(project, root);
-      if (shallDeleteLocalBranch) {
-        var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
-            .map(b -> b.equals(branchName))
-            .getOrElse(true);
-        if (slidOutBranchIsCurrent) {
-          LOG.debug("Skipping local branch deletion because it is equal to current branch");
-          getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+      GuiUtils.invokeLaterIfNeeded(
+          () -> deleteBranchIfRequired(anActionEvent, branchName, gitRepository.get(), project, shallDeleteLocalBranch),
+          ModalityState.NON_MODAL);
+
+    } else {
+      getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+    }
+  }
+
+  @UIEffect
+  private void deleteBranchIfRequired(AnActionEvent anActionEvent, String branchName, GitRepository gitRepository,
+      Project project, boolean shallDeleteLocalBranch) {
+    int okCancelDialogResult = Messages.NO;
```

Comment:
Also, maybe a nitpick... but once the dialog shows up, there's no way no cancel the slide-out... I've already ended up pressing `Esc` twice only to see that the action executed anyway :/ If there's some easy way to handle that, it'd be nice to have (but ok to put into another issue if that's more work ofc)